pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+          return str == null ? null : StrStatus.valueOf(str).getValue();
+        }
+      },
+      QPCR_PERCENTAGE_HUMAN("qpcr_percentage_human", "qPCR %"),
+      QUBIT_CONCENTRATION("qubit_concentration", "Qubit (ng/uL)"),
+      NANODROP_CONCENTRATION("nanodrop_concentration", "Nanodrop (ng/uL)"),
+      BARCODE("barcode", "Barcode"),
+      BARCODE_TWO("barcode_two", "Barcode Two"),
+      READ_LENGTH("read_length", "Read Length") {
+        private static final String PAIRED_KEY = "paired";
+
+        @Override
+        public String extractStringValueFrom(ResultSet rs) throws SQLException {
+          boolean paired = rs.getBoolean(PAIRED_KEY);
+          if (!rs.wasNull()) {
+            int readLength = rs.getInt(READ_LENGTH.getSqlKey());
+            if (!rs.wasNull()) {
+              return (paired ? "2x" : "1x") + readLength;
+            }
+          }
+          return null;
+        }
+      },
+      TARGETED_RESEQUENCING("targeted_sequencing", "Targeted Resequencing"),
+      SOURCE_TEMPLATE_TYPE("library_design_code", "Source Template Type"),
+      SUBPROJECT("subproject", "Sub-project"),
+      INSTITUTE("institute", "Institute"),
+      STAIN("stain", "Stain"),
+      SLIDES("slides", "Slides"),
+      DISCARDS("discards", "Discards"),
+      SLIDES_CONSUMED("slides_consumed", "Slides Consumed");
+
+      private final String sqlKey;
+      private final String attributeKey;
+
+      private AttributeKey(String sqlKey, String attributeKey) {
+        this.sqlKey = sqlKey;
+        this.attributeKey = attributeKey;
+      }
+
+      public String getSqlKey() {
+        return sqlKey;
+      }
+
+      public String getAttributeKey() {
+        return attributeKey;
+      }
+
+      /**
+       * Extracts the Attribute represented by this key from the result set. The column name within the ResultSet must match this.getKey()
+       * 
+       * @param rs
+       *          the ResultSet to extract the Attribute from
+       * @return
+       * @throws SQLException
+       */
+      public Attribute extractAttributeFrom(ResultSet rs) throws SQLException {
+        String val = extractStringValueFrom(rs);
+        return val == null ? null : makeAttribute(getAttributeKey(), extractStringValueFrom(rs));
+      };
+
+      /**
+       * Extracts the value belonging to this AttributeKey from a ResultSet
+       * 
+       * @param rs
+       *          ResultSet containing data to populate this field
+       * @return the value to associate with this AttributeKey; null if absent
+       * @throws SQLException
+       */
+      public String extractStringValueFrom(ResultSet rs) throws SQLException {
+        return rs.getString(getSqlKey());
+      }
+
+      private static Attribute makeAttribute(String name, String value) {
+        Attribute att = new DefaultAttribute();
+        att.setName(name);
+        att.setValue(value);
+        return att;
+      }
+
+      private static enum StrStatus {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this redundant "static" qualifier. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2786)